docs/core/reference/versioning.md

```diff
@@ -0,0 +1,72 @@
+# Chain Core Versioning
+
+## Summary
+
+Chain Core versioning falls into two categories: **package** versioning and **network** versioning.
+
+Package versioning applies to the pieces of software that work together to participate in a blockchain network. For example, the Chain Core Mac app and the Java SDK.
+
+Network versioning applies to the network API - the interface through which different Chain Cores on a network comunicate to exchange transactions and blocks. Any breaking changes to the network API interface or Chain protocol are reflected by a change to the network version.
+
+## Package Versioning
+The various packages of Chain Core each use a three number versioning scheme - **x.y.z**. The first two numbers indicate compatablity between packages. For example, if you are a running a version of the Chain Core Mac app whose first two numbers are **1.1**, then you should also use a version of an SDK whose first two numbers are also **1.1**.
+
+Sometimes we make bugfixes and minor feature updates to individual software packages. When this occurs, we update the third number in the version string for that package only. This number doesn't affect compatibility with other packages. For example, you can safely use version **1.1.1** of the Chain Core Mac App with version **1.1.3** of the Java SDK.
+
+### Scope
+
+The package versioning scheme covers a suite of individual software packages that make up Chain Core, including:
+
+- `cored`, the Chain Core server daemon
+- `corectl`, a CLI utility for operating Chain Core
+- `signerd`, the HSM signing server daemon
+- HSM firmware
+- Client API SDKs, such as the Java and Ruby SDKs
+- the Chain Core Mac app
+- the Chain Core Windows app
+- the Chain Core Docker image
+
+### Format
+
+Each release of a software package is assigned a version string composed of three numbers separated by periods, such as **1.0.1**.
+
+These numbers represent, in order:
+
+- **Major version**: significant product changes
+- **Minor version**: bugfixes, new features
+- **Build version**: package-specific bugfixes and features
+
+The major and minor versions are shared by all software packages in the Chain Core suite. If there is a change in the major or minor version, then there will be a new release of all software packages in the suite.
+
+The build version of a specific package may change independently of other packages.
+
+### Semantics
+
+This scheme has semantics that are unique to Chain Core, despite superficial similarities to other versioning schemes such as [Semantic Versioning](http://semver.org/).
+
+#### Compatibility between packages
+
+Software packages in the Chain Core suite are compatible if they share the same major and minor version. For example, if you are running version 1.1 of the Chain Core server, you should use version 1.1 of the SDK.
```

Comment:
This mostly makes sense, but a strict reading of this sentence means they would have to take downtime during an upgrade. It would be best if we can accommodate incremental rollouts, which implies having some minimal compatibility overlap between different values for the first two numbers in the version string.